sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPump.java

```diff
@@ -61,14 +65,23 @@ void setLastEnqueuedEventProperties(LastEnqueuedEventProperties lastEnqueuedEven
      */
     @Override
     public void close() {
+        // When client receive complete signal will call this method again in clean up method on scheduler thread,
+        // the first dispose will cause a timeout exception. Add a check avoid close twice.
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
         try {
             client.close();
         } catch (Exception error) {
             LOGGER.atInfo()
                 .addKeyValue(PARTITION_ID_KEY, partitionId)
                 .log("Exception occurred disposing of consumer client.", error);
         } finally {
-            scheduler.dispose();
+            try {
+                scheduler.disposeGracefully().block(Duration.ofSeconds(1));
+            } catch (IllegalStateException e) {
```

Comment:
What condition would cause this exception?

Reply:
If dispose timeout , `disposeGracefully()` will throw an IllegalStateException , the scheduler will still work. Users can combine `retry()` and etc. with `disposeGracefully()` if needed.  I just use block and  if dipose is timeout, call `dispose()` directly.